packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/container.tsx

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps } from '@reach/router';
+import CompleteResetPassword from '.';
+import { Integration, useAuthClient } from '../../../models';
+import LinkValidator from '../../../components/LinkValidator';
+import { LinkType } from '../../../lib/types';
+import { CreateCompleteResetPasswordLink } from '../../../models/reset-password/verification/factory';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import AppLayout from '../../../components/AppLayout';
+import { CardHeader } from '@material-ui/core';
+
+const CompleteResetPasswordContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+
+  // TODO: UX for this, FXA-8106
+  if (oAuthDataError) {
+    return (
+      <AppLayout>
+        <CardHeader
+          headingText="Unexpected error"
+          headingTextFtlId="auth-error-999"
+        />
+      </AppLayout>
+    );
+  }
+
+  // TODO: possibly rethink LinkValidator approach as it's a lot of layers with
```

Comment:
So as you know I am a fan of validating through models. Either via @bind, or the newly proposed way with the class-validator decorators which should be even better. 

Perhaps we repurpose the model validation for link validation? There's a few ways I can imagine this working, but one way is calling validate on the model in the container or maybe inside a validator (aka link validator) react component. Either way, the component could handle any errors coming from this model validtion operation, and render the appropriate UI.

Reply:
> Perhaps we repurpose the model validation for link validation?

That's what I was thinking 👍 I think `LinkValidator` was a good idea before realizing how much we needed something like the container components.